Userland/Libraries/LibWeb/HTML/HTMLFormElement.cpp

```diff
@@ -1012,4 +1040,66 @@ WebIDL::ExceptionOr<JS::Value> HTMLFormElement::named_item_value(FlyString const
     return node;
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#default-button
+FormAssociatedElement* HTMLFormElement::default_button()
+{
+    // A form element's default button is the first submit button in tree order whose form owner is that form element.
+    FormAssociatedElement* default_button = nullptr;
+
+    root().for_each_in_subtree([&](auto& node) {
+        auto* form_associated_element = dynamic_cast<FormAssociatedElement*>(&node);
```

Comment:
In LibGUI/LibCore we have `for_each_in_subtree_of_type`. I wonder if such a helper either exists on TreeNode, or if we should have one. Definitely a future thought.

Reply:
We have `for_each_in_subtree_of_type`, but can't be used in `FormAssociatedElement` as-is since `FormAssociatedElement` is not a `Node`. So the `static_cast`s used there (and in helpers like `verify_cast`) can't be used.